analyzer/src/funTest/kotlin/PubTest.kt

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer
+
+import com.here.ort.analyzer.managers.Pub
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.yamlMapper
+import com.here.ort.utils.normalizeVcsUrl
+import com.here.ort.utils.test.DEFAULT_ANALYZER_CONFIGURATION
+import com.here.ort.utils.test.DEFAULT_REPOSITORY_CONFIGURATION
+import com.here.ort.utils.test.USER_DIR
+import com.here.ort.utils.test.patchExpectedResult
+import io.kotlintest.shouldBe
+import io.kotlintest.matchers.startWith
+import io.kotlintest.shouldNotBe
+import io.kotlintest.specs.WordSpec
+
+import java.io.File
+
+class PubTest : WordSpec() {
+    private val projectsDir = File("src/funTest/assets/projects/synthetic/pub/").absoluteFile
+    private val projectsDirExternal = File("src/funTest/assets/projects/external/").absoluteFile
+    private val vcsDir = VersionControlSystem.forDirectory(projectsDir)!!
+    private val vcsUrl = vcsDir.getRemoteUrl()
+    private val vcsRevision = vcsDir.getRevision()
+
+    init {
+
+        "Pub" should {
+            "resolve dart http dependencies correctly" {
+                val workingDir = File(projectsDirExternal, "http")
+                val packageFile = File(workingDir, "pubspec.yaml")
+                var expectedResultFile = File(projectsDirExternal, "http-expected-output.yml");
+
+                val result = createPubForExternal().resolveDependencies(listOf(packageFile))[packageFile]
+                result!!.errors.size shouldBe 0
+
+                //expectedResultFile.writeText(yamlMapper.writeValueAsString(result));
+
+                val vcsPath = vcsDir.getPathToRoot(workingDir)
+                val expectedResult = patchExpectedResult(
+                    expectedResultFile,
+                    custom = Pair("pub-project", "pub-${workingDir.name}"),
+                    definitionFilePath = "$vcsPath/pubspec.yaml",
+                    url = normalizeVcsUrl(vcsUrl),
+                    revision = vcsRevision,
+                    path = vcsPath
+                )
+
+                yamlMapper.writeValueAsString(result) shouldBe expectedResult
+            }
+
+            "Resolve dependencies for a project with flutter correctly" {
+                val workingDir = File(projectsDir, "project-with-flutter")
+                val packageFile = File(workingDir, "pubspec.yaml")
+                var expectedResultFile = File(projectsDir.parentFile, "pub-expected-output-project-with-flutter.yml");
+
+                val result = createPub().resolveDependencies(listOf(packageFile))[packageFile]
+
+                if(!result!!.errors.isEmpty()) {
+                    print(result.errors.get(0).message);
+                }
+                //result.errors.size shouldBe 0
```

Comment:
Remove debugging code, also the check is implicitly done by the comparison below.